lib/benchee/conversion/memory.ex

```diff
@@ -46,6 +46,131 @@ defmodule Benchee.Conversion.Memory do
               }
   }
 
+  # Unit conversion functions
+
+  @doc """
+  Converts a value for a specified unit and converts it to the equivalent of another unit of measure.
+
+  ## Examples
+
+    iex> {value, unit} = Benchee.Conversion.Memory.convert({1024, %Benchee.Conversion.Unit{name: :kilobyte}}, %Benchee.Conversion.Unit{name: :megabyte})
+    iex> value
+    1.0
+    iex> unit.name
+    :megabyte
+  """
+  def convert({value, current_unit}, desired_unit) when current_unit == desired_unit do
+    {value, current_unit}
+  end
+  def convert({value, current_unit_atom}, desired_unit_atom) when is_atom(current_unit_atom) and is_atom(desired_unit_atom) do
+    current_value_unit = {value, unit_for(current_unit_atom)}
+    desired_unit = unit_for(desired_unit_atom)
+    convert(current_value_unit, desired_unit)
+  end
+
+  # Bytes conversions
+  def convert({value, %Unit{name: :byte}}, byte_unit = %Unit{name: :kilobyte}) do
```

Comment:
Just getting my thoughts down here - there is a rather similar shape to these `convert/2` functions, and it feels like there's maybe an additional abstraction that we might be able to make here eventually. However, it doesn't jump out to me right away, so I'm totally fine with the explicit version we have here for now.

Reply:
Heyho,

that's my main point here as well. And the shape is rather easy, I think we can delete 90% of the code here. Each unit has a `magnitude` field. So all `convert` needs to do is:

```
def convert({value, %Unit{magnitude: current_magnitude}}, desired_unit = %Unit{magnitude: desired_magnitude}) do
  multiplier = current_magnitude / desired_magnitude
  {value * multiplier, desired_unit}
end
```

That should cover all of it imo or what am I missing? Could be implemented on `Scale` itself and then just called from the implementing module.

Reply:
Yeah I felt the code as I had it "smelled" a bit funny and could be further abstracted. Just couldn't put my finger on the implementation. I'll make the changes in the next commit

Reply:
@PragTob I feel that the conversion methods could belong the Memory module, they do convert memory from one magnitude to another after all. Currently the method is called only from the ```System``` module as well. I could totally be missing something since I'm new to the codebase, please let me know if I am off base here!